### PR TITLE
For #381: Evade buff effects

### LINKED ISSUE #381: Implement Evade Buff
See Glossary.

Size: 2 units

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #381: Evade buff effects
- Added `Evade` effect, which evades an attack
- Created `EvadeAttacks(int quantity)`, which alows the target to evade `quantity` attacks
- Created tests for `Evade`

### CODE REVIEW

File: src/ProjectNeon/Assets/Features/Effects/Evade.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
+public class Evade : Effect
+{
+    private Member _performer;
+    private Target _effectTarget;
+
+    public void Apply(Member source, Target target)
+    {
+        _performer = source;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/Evade.cs]: Why are we storing these member fields, when they could be directly passed to the evaluation method?

File: src/ProjectNeon/Assets/Features/Effects/EvadeAttacks.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using UnityEngine;
+using UnityEditor;
+
+public sealed class EvadeAttacks : Effect
+{
+    private Effect _effect;
+
+    public EvadeAttacks(int quantity)
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/EvadeAttacks.cs]: Do we need this alias class?

If we do, then should be following proper `ctor` EO rules. Save the argument. Don't hide any factory code.

File: src/ProjectNeon/Assets/Z_Tests/Features/Effects/EvadeTests.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿
+using NUnit.Framework;
+using UnityEngine;
+
+public sealed class EvadeTests
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Z_Tests/Features/Effects/EvadeTests.cs]: We need to test to ensure that the second attack is NOT evaded.

File: src/ProjectNeon/Assets/Features/Effects/Evade.cs (hunk 1 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Effects/Evade.cs]: @SilasReinagel Didn't get it. `Evade` effect is triggered on an `Attack`, so we need to keep these fields to know who performed the `Evade` buff (who added this evade effect to the stack) and who is being benefited by it

File: src/ProjectNeon/Assets/Features/Effects/EvadeAttacks.cs (hunk 2 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Effects/EvadeAttacks.cs]: @SilasReinagel Removed it

File: src/ProjectNeon/Assets/Z_Tests/Features/Effects/EvadeTests.cs (hunk 3 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Z_Tests/Features/Effects/EvadeTests.cs]: @SilasReinagel Actually these tests are about the `Evade` effect, not the `Recurrent`. I've refactored them to use only the `Evade` implementation

File: src/ProjectNeon/Assets/Features/Effects/Evade.cs (hunk 1 above)
Comment (SilasReinagel) [reply src/ProjectNeon/Assets/Features/Effects/Evade.cs]: This issue is still open.

What part is confusing? We don't need any fields in the class.

Comment (SilasReinagel) [reply src/ProjectNeon/Assets/Features/Effects/Evade.cs]: @paulodamaso Forgot to tag you.

Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Effects/Evade.cs]: @SilasReinagel `Evade` works as any other triggered `Effect`: first, we apply it, then we execute it when triggered. If someone have the `Evade` buff, the evade effect is active on him (preferrably using the decorators that limit events by turns or by repetitions). The `Evade` effects is triggered before an attack happens, so we need to check if we are applying it to the correct target. So we need to save the original target received on `Apply`

**Pull request merged**